src/Microsoft.TemplateEngine.Cli/Commands/NewCommand.cs

```diff
@@ -182,40 +185,40 @@ protected override async Task<NewCommandStatus> ExecuteAsync(NewCommandArgs args
             }
 
             using TemplatePackageManager templatePackageManager = new TemplatePackageManager(environmentSettings);
+            var hostSpecificDataLoader = new HostSpecificDataLoader(environmentSettings);
 
             if (string.IsNullOrWhiteSpace(args.ShortName))
             {
                 TemplateListCoordinator templateListCoordinator = new TemplateListCoordinator(
                     environmentSettings,
                     templatePackageManager,
-                    new HostSpecificDataLoader(environmentSettings),
+                    hostSpecificDataLoader,
                     TelemetryLogger);
 
-                //TODO: we need to await, otherwise templatePackageManager will be disposed.
                 return await templateListCoordinator.DisplayCommandDescriptionAsync(args, default).ConfigureAwait(false);
             }
 
             var templates = await templatePackageManager.GetTemplatesAsync(context.GetCancellationToken()).ConfigureAwait(false);
-            var template = templates.FirstOrDefault(template => template.ShortNameList.Contains(args.ShortName));
+            var templateGroups = TemplateGroup.FromTemplateList(CliTemplateInfo.FromTemplateInfo(templates, hostSpecificDataLoader));
 
-            if (template == null)
-            {
-                Reporter.Error.WriteLine($"Template {args.ShortName} doesn't exist.");
-                return NewCommandStatus.NotFound;
-            }
+            //TODO: decide what to do if there are more than 1 group.
```

Comment:
I think we dicussed this, if 2 groups have same ShortName, for both groups instead of shortname we display GroupdId.